oggm/workflow.py

```diff
@@ -369,3 +371,78 @@ def inversion_tasks(gdirs):
 
     # Filter
     execute_entity_task(tasks.filter_inversion_output, gdirs)
+
+
+def merge_glacier_tasks(glcdf, main_rgi_ids, filename='climate_monthly',
+                        input_filesuffix=''):
+    """Shortcut function: run all tasks to merge tributaries to a main glacier
+
+    TODO - Automatic search for tributary glaciers
+    TODO - Every tributary should only be used once
+
+    Parameters
+    ----------
+    glcdf: geopandas.GeoDataFrame
+        which contains the main glaciers and the candidates
+    main_rgi_ids: list of str
+        RGI IDs of the main glaciers of interest. Must be in glcdf.
+    filename: str
+        Baseline climate file
+    input_filesuffix: str
+        Filesuffix to the climate file
+
+    Returns
+    -------
+    merged_gdirs: list of merged GlacierDirectories
+
+    """
+    # init and preprocess all glaciers
+    gdirs_all = init_glacier_regions(glcdf)
+    gis_prepro_tasks(gdirs_all)
+
+    # make sure rgi_ids are iteratable
+    main_rgi_ids = utils.tolist(main_rgi_ids)
+
+    # split main glaciers from candidates
+    maindf = glcdf.loc[glcdf.RGIId.isin(main_rgi_ids)].copy()
+    gdirs_main = [gd for gd in gdirs_all if gd.rgi_id in main_rgi_ids]
+
+    # preprocess the main glaciers to the end
+    climate_tasks(gdirs_main)
+    inversion_tasks(gdirs_main)
+    execute_entity_task(tasks.init_present_time_glacier, gdirs_main)
+
+    # find true tributary glaciers
+    tributaries = execute_entity_task(
+        centerlines.intersect_downstream_lines,
+        gdirs_main, candidates=gdirs_all)
+
+    # make one dictionary and a list of all gdirs for further preprocessing
+    tribs_dict = {}
+    gdirs_tribs = []
+    for trb in tributaries:
+        tribs_dict.update(trb)
+        for gd in trb.values():
+            gdirs_tribs += gd
+
+    # check if all tributaries are only used once
+    rgiids = [gd.rgi_id for gd in gdirs_tribs]
+    if not len(np.unique(rgiids)) == len(rgiids):
+        raise RuntimeError('Every tributary glacier should only be used once!')
+
+    # preprocess the tributaries to the end
+    climate_tasks(gdirs_tribs)
+    inversion_tasks(gdirs_tribs)
+    execute_entity_task(tasks.init_present_time_glacier, gdirs_tribs)
```

Comment:
I guess you are doing this to preprocess only the glaciers you really need. I can see why this is useful, but it feels quite redundant.